packages/wallet/src/services/DelegationTracker/DelegationTracker.ts

```diff
@@ -68,6 +90,95 @@ export const certificateTransactionsWithEpochs = (
     )
   );
 
+export const createDelegationPortfolioTracker = ({
+  rewardAccounts$,
+  knownAddresses$,
+  utxoTracker
+}: DelegationPortfolioTrackerProps): Observable<Cip17Pool[]> => {
+  const balanceAllUtxosTracker = createBalanceByAddressTracker(utxoTracker);
+
+  const delegatedAccounts$ = rewardAccounts$.pipe(
+    map((rewardsAccounts) =>
+      rewardsAccounts.filter(
+        (account) => account.keyStatus === Cardano.StakeKeyStatus.Registered && account.delegatee?.nextNextEpoch
+      )
+    )
+  );
+
+  const hydratedDelegatedAccounts$ = delegatedAccounts$.pipe(
+    withLatestFrom(knownAddresses$),
+    map(([delegatedAccounts, knownAddresses]) =>
+      delegatedAccounts.map((delegatedAccount) => {
+        const groupedAddress = knownAddresses.find(
+          (knownAddr) => knownAddr.rewardAccount === delegatedAccount.address
+        )!;
+        return {
+          balance: createBalanceByAddressTracker(utxoTracker, groupedAddress.address),
```

Comment:
There could potentially be multiple addresses for a single reward account
```suggestion
        const groupedAddresses = knownAddresses.filter(
          (knownAddr) => knownAddr.rewardAccount === delegatedAccount.address
        )!;
        return {
          balance: createBalanceByAddressesTracker(utxoTracker, groupedAddresses),
```
Alternatively could do `createBalanceByRewardAccountTracker`

Reply:
great catch! thanks

Reply:
Fixed in [4bb58e4](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/4bb58e415d7e58c437939064515414896695ba6a) & [b59a6c0](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/b59a6c077c5e4b9dfb7f77526cbd6b97b8204706)